pkg/plugins/plugins_test.go

```diff
@@ -23,11 +23,76 @@ import (
 
 	"gotest.tools/v3/assert"
 
-	"github.com/apache/yunikorn-core/pkg/mock"
+	"github.com/apache/yunikorn-scheduler-interface/lib/go/si"
 )
 
+type NoPluginImplemented struct{}
+
+type RMPluginImplemented struct{}
+
+type StateDumpImplemented struct{}
+
+type AllPluginsImplemented struct {
+	RMPluginImplemented
```

Comment:
why not reuse `mock.ResourceManagerCallback`?

Reply:
That will cause circular imports when we do that in other pieces of the code besides the fact that we need different implementations for each interface so we can register one interface at a time.